python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py

```diff
@@ -188,3 +189,8 @@ def get_asset_key(self, dbt_resource_props: Mapping[str, Any]) -> AssetKey:
             return base_key.with_prefix(self._source_asset_key_prefix)
         else:
             return base_key.with_prefix(self._asset_key_prefix)
+
+
+@dataclass
+class DbtManifestWrapper:
+    manifest: Mapping[str, Any]
```

Comment:
how does this wrapper prevent the problem?

Reply:
We just tracked down the reason our mem usage was so high to this object so I think it likely is not fixing the real problem.

Reply:
@pablo-statsig From the PR description:

> This was because we serialized the contents of the dbt project manifest as metadata on each of the software-defined assets generated by the integration.

The original problem was that the manifest was being stored on each serialized representation of the software-defined asset. This means that when you viewed a dbt model in the Dagster UI, a metadata entry with the contents of the dbt manifest was showing up for each dbt SDA. By wrapping the manifest in a non-serializable object, we prevent this from happening.

Outside of the serialization problem, we still need to hold a reference to the dbt manifest from the user to define their software-defined assets and add metadata to their computations.

Depending on how the user reads the manifest in their user code, this might introduce some unnecessary memory costs. For example, if the user had multiple `@dbt_assets` definitions from the same manifest, but used a `Path` reference for the manifest instead of reading the manifest into memory as a `dict`, we would hold multiple copies of the manifest rather than having multiple references to the same copy: https://github.com/dagster-io/dagster/issues/17980.

This was resolved in https://github.com/dagster-io/dagster/pull/17996 (`dagster-dbt==0.21.9`).

If you are seeing other issues, please file a separate issue with the behavior that you're seeing. Thanks!